enhancements/network/upi-node-ip-selection.md

```diff
@@ -0,0 +1,218 @@
+---
+title: upi-node-ip-selection
+authors:
+  - "@pliu"
+reviewers: 
+  - "@cybertron"
+  - "@jcpowermac"
+  - "@patrickdillon"
+approvers:
+  - "@danwinship"
+  - "@trozet"
+api-approvers: 
+  - "None"
+creation-date: 2023-02-08
+last-updated: 2023-02-16
+tracking-link:
+  - https://issues.redhat.com/browse/NP-687
+see-also:
+  - "/enhancements/network/ip-interface-selection"
+replaces:
+superseded-by:
+---
+
+# Node IP Selection UPI
+
+## Summary
+
+When Openshift is deployed on user-provisioned infrastructure, and there are
+more than one NIC available on the hosts. We need a reliable way to guarantee
+all the openshift components can use the correct IP/interface.
+
+## Motivation
+
+Today, we do not run the nodeip-configuration service on vSphere UPI. It means
+we don't have the mechanism to ensure IP/interface selection will return the
+same result for following components:
+
+- Node IP (Kubelet and CRIO)
+- configure-ovs
+- resolv-prepender
+- Keepalived
+- Etcd
+
+Those components may select different IP/interface as the nodeIP, when there are
+multiple NICs in a node, but no default gateway is configured on any of them.
+
+For other UPI platforms, although the nodeip-configuration service is enabled.
+It runs in a mode that it will only return the IP/interface with the default
+route. The assumption is that the interface with the default route shall have
+the nodeIP. However, this assumption is not always true.
+
+Therefore we need to have a generic way for all the UPI platforms. So that users
+can specify the node IP/interface selection explicitly.
+
+### User Stories
+
+As a deployer, I want to explicitly specify the UPI cluster machine network to a
```

Comment:
We don't really distinguish between UPI & IPI in code--at least in the installer. This change would affect the infrastructure object in IPI installs as well as UPI.